cpp/src/Slice/Grammar.y

```diff
@@ -150,25 +226,23 @@ meta_data
 // ----------------------------------------------------------------------
 definitions
 // ----------------------------------------------------------------------
-: global_meta_data
+: definitions global_meta_data
 {
-    StringListTokPtr metaData = StringListTokPtr::dynamicCast($1);
+    StringListTokPtr metaData = StringListTokPtr::dynamicCast($2);
     if(!metaData->v.empty())
     {
         unit->addGlobalMetaData(metaData->v);
     }
 }
-definitions
-| meta_data definition
+| definitions meta_data definition
 {
-    StringListTokPtr metaData = StringListTokPtr::dynamicCast($1);
-    ContainedPtr contained = ContainedPtr::dynamicCast($2);
+    StringListTokPtr metaData = StringListTokPtr::dynamicCast($2);
+    ContainedPtr contained = ContainedPtr::dynamicCast($3);
     if(contained && !metaData->v.empty())
     {
         contained->setMetaData(metaData->v);
     }
 }
-definitions
```

Comment:
Left-recursive rules should always be preferred to right-recursive rules, since the parser combines tokens in order. There's probably more places this needs to be fixed in the parser.
```
foo foo_list	// has to wait until the very end to combine foos into a foo_list. Stack will need N elements.
foo_list foo    // can add foos into the list as it goes along. Stack only needs 2 elements
```